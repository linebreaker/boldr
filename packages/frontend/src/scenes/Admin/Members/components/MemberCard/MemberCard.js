/* @flow */
import React from 'react';
import styled from 'styled-components';
import Button from '@boldr/ui/Button';
import Avatar from '@boldr/ui/Avatar';

const MemListItem = styled.li`
  width: 100%;
  display: flex;
  margin-top: 10px;
  padding: 20px;
  align-items: center;
  border-radius: 2px;
  background: #fff;
  box-shadow: 0 2px 1px rgba(170, 170, 170, 0.25);
`;

const RightSide = styled.div`
  position: relative;
  align-items: flex-end;
`;
const UserInfoList = styled.ul`
  display: inline-flex;
  align-items: center;
  list-style-type: none;
  justify-content: space-between;
`;
const UserInfoListItem = styled.li`
  padding-left: 1em;
  padding-right: 1em;
`;
const LeftSide = styled.div`
  align-items: flex-start;
  width: 90%;
`;

type Props = {
  user: Object,
  handleToggle: () => void,
  username: string,
  lastName: string,
  firstName: string,
  roleName: string,
  id: string,
  email: string,
  avatarUrl: string,
};

const MemberCard = (props: Props) => {
  const { user, handleToggle, username, roleName, avatarUrl, lastName, firstName, email } = props;
  return (
    <MemListItem>
      <LeftSide>
        <UserInfoList>
          <UserInfoListItem>
            <Avatar src={avatarUrl} role="presentation" />
          </UserInfoListItem>
          <UserInfoListItem>
            {username}
          </UserInfoListItem>
          <UserInfoListItem>
            {roleName}
          </UserInfoListItem>
          <UserInfoListItem>
            {firstName} {lastName}
          </UserInfoListItem>
          <UserInfoListItem>
            {email}
          </UserInfoListItem>
        </UserInfoList>
      </LeftSide>
      <RightSide>
        <Button onClick={() => handleToggle(user)} kind="primary">
          Edit
        </Button>
      </RightSide>
    </MemListItem>
  );
};

export default MemberCard;
