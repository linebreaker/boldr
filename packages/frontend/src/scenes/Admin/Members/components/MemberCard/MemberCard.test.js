import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MemberCard from './MemberCard';

const userData = {
  id: 'abad-12dc-3dfd',
  username: 'joe',
  email: 'test@test.com',
  roles: [{ id: 1, name: 'admin' }],
  avatarUrl: 'http://img.com/img.png',
};
const id = 'abad-12dc-3dfd';

describe('<MemberCard />', () => {
  it('should render the user props', () => {
    const handleToggle = jest.fn();
    const renderedComponent = shallow(
      <MemberCard
        user={userData}
        key={1}
        username="joe"
        handleToggle={handleToggle}
        id="abad-12dc-3dfd"
        avatarUrl="http://img.com/img.png"
        email="test@test.com"
        roleName="admin"
      />,
    );
    expect(shallowToJson(renderedComponent)).toMatchSnapshot();
  });
});
