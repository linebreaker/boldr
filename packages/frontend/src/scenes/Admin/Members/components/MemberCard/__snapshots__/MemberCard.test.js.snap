// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemberCard /> should render the user props 1`] = `
<styled.li>
  <styled.div>
    <styled.ul>
      <styled.li>
        <Avatar
          role="presentation"
          src="http://img.com/img.png"
          suffixes={
            Array [
              "red",
              "pink",
              "purple",
              "deep-purple",
              "indigo",
              "blue",
              "light-blue",
              "cyan",
              "teal",
              "green",
              "light-green",
              "lime",
              "yellow",
              "amber",
              "orange",
              "deep-orange",
              "brown",
              "grey",
              "blue-grey",
            ]
          }
        />
      </styled.li>
      <styled.li>
        joe
      </styled.li>
      <styled.li>
        admin
      </styled.li>
      <styled.li>
         
      </styled.li>
      <styled.li>
        test@test.com
      </styled.li>
    </styled.ul>
  </styled.div>
  <styled.div>
    <Button
      block={false}
      bordered={true}
      className=""
      disabled={false}
      htmlType="button"
      kind="primary"
      loading={false}
      onClick={[Function]}
      outline={false}
      prefix="boldrui"
      size="medium"
    >
      Edit
    </Button>
  </styled.div>
</styled.li>
`;
